### PR TITLE
RAC-31: Update versioning normalizer for quantified associations

### PR DESCRIPTION
**Pull request by StevenVAIDIE**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | Todo
| Added legacy Behats               | Todo
| Added acceptance tests            | Todo
| Added integration tests           | Todo
| Changelog updated                 | Todo
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Akeneo/Pim/Enrichment/Component/Product/Normalizer/Versioning/EntityWithQuantifiedAssociations/QuantifiedAssociationsNormalizer.php [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+     * @param EntityWithQuantifiedAssociationsInterface $entityWithQuantifiedAssociations
+     */
+    public function normalize($entityWithQuantifiedAssociations, $format = null, array $context = [])
+    {
+        $quantifiedAssociationsNormalized = $entityWithQuantifiedAssociations->normalizeQuantifiedAssociations();
+
+        $results = [];
+        foreach ($quantifiedAssociationsNormalized as $associationTypeCode => $linkTypes) {
+            foreach ($linkTypes['products'] as $quantifiedLink) {
+                $flatAssociationKey = $associationTypeCode . '-products-' . $quantifiedLink['identifier'];
```
Comment (ValentinMumble) [inline src/Akeneo/Pim/Enrichment/Component/Product/Normalizer/Versioning/EntityWithQuantifiedAssociations/QuantifiedAssociationsNormalizer.php:25]:
```suggestion
                $flatAssociationKey = sprintf('%s-products-%s', $associationTypeCode, $quantifiedLink['identifier']);
```

File: src/Akeneo/Pim/Enrichment/Component/Product/Normalizer/Versioning/EntityWithQuantifiedAssociations/QuantifiedAssociationsNormalizer.php [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+                $results[$flatAssociationKey] = $quantifiedLink['quantity'];
+            }
+
+            foreach ($linkTypes['product_models'] as $quantifiedLink) {
+                $flatAssociationKey = $associationTypeCode . '-product_models-' . $quantifiedLink['identifier'];
```
Comment (ValentinMumble) [inline src/Akeneo/Pim/Enrichment/Component/Product/Normalizer/Versioning/EntityWithQuantifiedAssociations/QuantifiedAssociationsNormalizer.php:30]:
```suggestion
                $flatAssociationKey = sprintf('%s-product_models-%s', $associationTypeCode, $quantifiedLink['identifier']);
```

File: src/Akeneo/Pim/Enrichment/Component/Product/Normalizer/Versioning/EntityWithQuantifiedAssociations/QuantifiedAssociationsNormalizer.php [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+     *
+     * @param EntityWithQuantifiedAssociationsInterface $entityWithQuantifiedAssociations
+     */
+    public function normalize($entityWithQuantifiedAssociations, $format = null, array $context = [])
+    {
+        $quantifiedAssociationsNormalized = $entityWithQuantifiedAssociations->normalizeQuantifiedAssociations();
+
+        $results = [];
+        foreach ($quantifiedAssociationsNormalized as $associationTypeCode => $linkTypes) {
+            foreach ($linkTypes['products'] as $quantifiedLink) {
```
Comment (juliensnz) [inline src/Akeneo/Pim/Enrichment/Component/Product/Normalizer/Versioning/EntityWithQuantifiedAssociations/QuantifiedAssociationsNormalizer.php:24]: I would prefer an array_map here with an array merge at the end. But it's ok like that also

Comment (StevenVAIDIE) [reply src/Akeneo/Pim/Enrichment/Component/Product/Normalizer/Versioning/EntityWithQuantifiedAssociations/QuantifiedAssociationsNormalizer.php]: I did it in : https://github.com/akeneo/pim-community-dev/pull/12202/commits/9ea3569c0a85049c98736c810e8b663089dbd19a.

Please tell me if you were thinking of something else

**Pull request merged**